tasks/tests/integration/test_upload_e2e.py

```diff
@@ -0,0 +1,194 @@
+
+    redis_key = f"uploads/{repoid}/{commitid}"
+    uploads = redis.lists.setdefault(redis_key, [])
+    uploads.append(json.dumps({"reportid": report_id, "url": written_path}))
+
+
+def lines(lines: Iterable[tuple[int, ReportLine]]) -> list[tuple[int, int]]:
+    return list(((lineno, line.coverage) for lineno, line in lines))
+
+
+@pytest.mark.integration
+@pytest.mark.django_db()
+def test_full_upload(
+    dbsession: Session,
+    mocker,
+    mock_redis,
+    mock_repo_provider,
+    mock_storage,
+    mock_configuration,
+):
+    hook_session(dbsession, mocker)
+
+    # to not close the session after each task
+    mocker.patch("tasks.base.BaseCodecovTask.wrap_up_dbsession")
+    # avoid some calls that are failing for unknown reasons:
+    mocker.patch("tasks.upload.UploadTask.possibly_setup_webhooks", return_value=True)
+    mocker.patch(
+        "tasks.upload.fetch_commit_yaml_and_possibly_store", return_value=UserYaml({})
+    )
+    mocker.patch(
+        "tasks.upload.get_repo_provider_service",
+        return_value=mock_repo_provider,
+    )
+    mocker.patch(
+        "tasks.upload_processor.get_repo_provider_service",
+        return_value=mock_repo_provider,
+    )
+    mocker.patch(
+        "tasks.notify.get_repo_provider_service",
+        return_value=mock_repo_provider,
+    )
+    # for some reason, the `archive` down below still does not contain the files+sessions file?
+    mock_configuration.set_params(
+        {
+            "setup": {
+                "save_report_data_in_storage": {
+                    "only_codecov": False,
+                    "commit_report": True,
+                    "report_details_files_array": True,
+                },
+            }
+        }
+    )
+
+    repository = RepositoryFactory.create()
+    dbsession.add(repository)
+    dbsession.flush()
+
+    repoid = repository.repoid
+    commitid = uuid4().hex
+    commit = CommitFactory.create(repository=repository, commitid=commitid)
+    dbsession.add(commit)
+    dbsession.flush()
+
+    archive_service = ArchiveService(repository)
+    do_upload = partial(
+        write_raw_upload,
+        mock_redis,
+        archive_service,
+        repoid,
+        commitid,
+    )
+
+    do_upload(b"""
+a.rs
+<<<<<< network
+# path=coverage.lcov
+SF:a.rs
+DA:1,1
+end_of_record
+""")
+    do_upload(b"""
+a.rs
+<<<<<< network
+# path=coverage.lcov
+SF:a.rs
+DA:2,2
+end_of_record
+""")
+    do_upload(b"""
+b.rs
+<<<<<< network
+# path=coverage.lcov
+SF:b.rs
+DA:1,3
+end_of_record
+""")
+
+    with run_tasks():
+        upload_task.apply_async(
+            kwargs={
+                "repoid": repoid,
+                "commitid": commitid,
+            }
+        )
+
+    archive = mock_storage.storage["archive"]
+    repo_hash = ArchiveService.get_archive_hash(repository)
+    raw_chunks_path = f"v4/repos/{repo_hash}/commits/{commitid}/chunks.txt"
+    assert raw_chunks_path in archive
+
+    # TODO: where is the `report_json` ???
```

Comment:
It is usually saved in storage... in `commit._report_json_path` maybe it's the pointer to the data?